src/main/java/bio/terra/service/tabulardata/WalkRelationship.java

```diff
@@ -27,21 +36,27 @@ public enum WalkDirection {
   private int fromIndex; // index of table we are walking from
   private int toIndex; // index of table we are walking to
 
+  private boolean strictRelationshipDirection;
+
   public static List<WalkRelationship> ofAssetSpecification(AssetSpecification assetSpecification) {
     List<WalkRelationship> walklist = new ArrayList<>();
     for (AssetRelationship assetRelationship : assetSpecification.getAssetRelationships()) {
-      walklist.add(WalkRelationship.ofAssetRelationship(assetRelationship));
+      walklist.add(
+          WalkRelationship.ofAssetRelationship(
+              assetRelationship, assetSpecification.isFollowStrictDirection()));
     }
     return walklist;
   }
 
-  public static WalkRelationship ofAssetRelationship(AssetRelationship assetRelationship) {
+  public static WalkRelationship ofAssetRelationship(
+      AssetRelationship assetRelationship, boolean strictRelationshipDirection) {
     Relationship datasetRelationship = assetRelationship.getDatasetRelationship();
     return new WalkRelationship()
         .fromColumn(datasetRelationship.getFromColumn())
         .fromTable(datasetRelationship.getFromTable())
         .toColumn(datasetRelationship.getToColumn())
-        .toTable(datasetRelationship.getToTable());
+        .toTable(datasetRelationship.getToTable())
+        .setStrictRelationshipDirection(strictRelationshipDirection);
```

Comment:
Did you consider making strict an aspect of the relationship itself? I guess since we want to enforce it for all relationships in an asset it would be a bit redundant, but maybe that's conceptually were it belongs.

Reply:
(we discussed this in person) But, I think that would be a good long term solution. For now, we're not propagating this parameter up to the user so we can test out its usefulness.